src/isomorphic/classic/element/ReactElement.js

```diff
@@ -163,12 +163,17 @@ ReactElement.createElement = function(type, config, children) {
 
   if (config != null) {
     if (__DEV__) {
+      var displayName = typeof type === 'function' ?
+        (type.displayName || type.name || 'Unknown') :
```

Comment:
`'Unkown'` is a weird default, imagine getting an error message like:

`React.createElement(...): Expected props argument to be a plain object. Properties defined in its prototype chain will be ignored. Unknown.`

Why not make it an empty string and leave as-is by default?

I'd also change the error message instead of appending the name, what about something like this:

```JS
'React.createElement(...): Expected props argument of ' + displayName +
' to be a plain object. Properties defined in its prototype chain will be ignored.'
```